nipype/interfaces/traits_extension.py

```diff
@@ -218,6 +223,68 @@ def __init__(self, value='', auto_set=False, entries=0,
         super(Directory, self).__init__(value, auto_set, entries, exists,
                                         **metadata)
 
+# lists of tuples
+# each element consists of :
+# - uncompressed (tuple[0]) extension
+# - compressed (tuple[1]) extension
+img_fmt_types = {
+        'nifti1': [('.nii', '.nii.gz'),
+                   (('.hdr', '.img'), ('.hdr', '.img.gz'))],
+        'mgh': [('.mgh', '.mgz'), ('.mgh', '.mgh.gz')],
+        'nifti2': [('.nii', '.nii.gz'),
+                   (('.hdr', '.img'), ('.hdr', '.img.gz'))],
+        'cifti2': [('.nii', '.nii.gz'),
+                   (('.hdr', '.img'), ('.hdr', '.img.gz'))],
+        'DICOM': [('.dcm', '.dcm'), ('.IMA', '.IMA'), ('.tar', '.tar.gz')],
+        'nrrd': [('.nrrd', 'nrrd'), ('nhdr', 'nhdr')],
+        'afni': [('.HEAD', '.HEAD'), ('.BRIK', '.BRIK')]
+        }
+
+class ImageFile(File):
+    """ Defines a trait of specific neuroimaging files """
+
+    def __init__(self, value='', filter=None, auto_set=False, entries=0,
+                 exists=False, types=[], compressed=True, extensions=[],
+                 **metadata):
+        """ Trait handles neuroimaging files.
+
+        Parameters
+        ----------
+        types : list
+            Strings of file format types accepted
+        compressed : boolean
+            Indicates whether the file format can compressed
+        """
+        self.types = types
+        self.compressed = compressed
+        self.exts = extensions
+        super(ImageFile, self).__init__(value, filter, auto_set, entries,
+                                        exists, **metadata)
+
+    def grab_exts(self, exts=[]):
+        for fmt in self.types:
+            if fmt in img_fmt_types:
+                exts.extend(sum([[u for u in y[0]] if isinstance(y[0], tuple)
+                            else [y[0]] for y in img_fmt_types[fmt]], []))
+                if self.compressed:
```

Comment:
`compressed=True` should allow for compressed and normal. perhaps we should change `compressed` to `allow_compressed`